testproject/Assets/Tests/Runtime/MultiprocessRuntime/BaseMultiprocessTests.cs

```diff
@@ -17,40 +22,80 @@ public class MultiprocessTestsAttribute : CategoryAttribute
 
         /// <summary>
         /// Implement this to specify the amount of workers to spawn from your main test runner
         /// TODO there's a good chance this will be re-factored with something fancier once we start integrating with bokken
         /// </summary>
-        protected abstract int WorkerCount { get; }
+        protected string[] platformList { get; set; }
 
         private const string k_FirstPartOfTestRunnerSceneName = "InitTestScene";
 
         // Since we want to additively load our BuildMultiprocessTestPlayer.MainSceneName
         // We want to keep a reference to the
         private Scene m_OriginalActiveScene;
 
+        // As an alternative to polling ConnectedClients Count we can store a
+        // collection of connected clients as reported by the client connect
+        // and client disconnect callbacks
+        protected List<ulong> m_ConnectedClientsList;
+
+        public bool IsSceneLoading = false;
+
         [OneTimeSetUp]
         public virtual void SetupTestSuite()
         {
-            MultiprocessLogger.Log("Running SetupTestSuite - OneTimeSetup");
-            if (IgnoreMultiprocessTests)
+            m_ConnectedClientsList = new List<ulong>();
+
+            platformList = MultiprocessOrchestration.GetRemotePlatformList();
+            if (platformList == null)
+            {
+                WorkerCount = 1;
+                m_LaunchRemotely = false;
+            }
+            else
+            {
+                WorkerCount = platformList.Length;
+                m_LaunchRemotely = true;
+                foreach (var platformListItem in platformList)
+                {
+                    MultiprocessLogger.Log($"Platform: {platformListItem}");
+                }
+            }
+
+            MultiprocessLogger.Log($"BaseMultiprocessTests - Running SetupTestSuite - LaunchRemotely {m_LaunchRemotely} MultiprocessOrchestration.ShouldRunMultiMachineTests() {MultiprocessOrchestration.ShouldRunMultiMachineTests()}");
+            if (m_LaunchRemotely && !MultiprocessOrchestration.ShouldRunMultiMachineTests())
             {
-                Assert.Ignore("Ignoring tests under UTR. For testing, include the \"-bypassIgnoreUTR\" command line parameter.");
+                Assert.Ignore($"Ignoring tests that require bokken for multimachine testing since as enableMultiMachineTesting Editor command line option not specified");
             }
 
             if (IsPerformanceTest)
             {
                 Assert.Ignore("Performance tests should be run from remote test execution on device (this can be ran using the \"run selected tests (your platform)\" button");
             }
-            MultiprocessLogger.Log($"Currently active scene {SceneManager.GetActiveScene().name}");
-            var currentlyActiveScene = SceneManager.GetActiveScene();
 
+            if (!MultiprocessOrchestration.ShouldRunMultiprocessTests())
```

Comment:
Is this WIP (if so you might want to decorate it with a note)?  Otherwise, what does it do?

Reply:
Ah, it's a little bit of legacy that needs clean up, I'll address this shortly.